services/command-router/src/main/java/org/eclipse/hono/commandrouter/CommandConsumerFactory.java

```diff
@@ -13,49 +13,20 @@
 
 package org.eclipse.hono.commandrouter;
 
-import java.time.Duration;
-
 import org.eclipse.hono.client.CommandTargetMapper;
-import org.eclipse.hono.client.ConnectionLifecycle;
-import org.eclipse.hono.client.HonoConnection;
-import org.eclipse.hono.client.SendMessageSampler;
 import org.eclipse.hono.client.ServiceInvocationException;
-import org.eclipse.hono.commandrouter.impl.amqp.ProtonBasedCommandConsumerFactoryImpl;
+import org.eclipse.hono.util.Lifecycle;
 
 import io.opentracing.SpanContext;
 import io.vertx.core.Future;
 
 /**
- * A factory for creating clients for the <em>AMQP 1.0 Messaging Network</em> to
- * receive commands and send responses.
+ * A factory for creating consumers of command &amp; control messages.
  */
-public interface CommandConsumerFactory extends ConnectionLifecycle<HonoConnection> {
-
-    /**
-     * Creates a new factory for an existing connection.
-     *
-     * @param connection The connection to the AMQP network.
-     * @return The factory.
-     * @throws NullPointerException if connection or gatewayMapper is {@code null}.
-     */
-    static CommandConsumerFactory create(final HonoConnection connection) {
-        return create(connection, SendMessageSampler.Factory.noop());
-    }
+public interface CommandConsumerFactory extends Lifecycle {
```

Comment:
Regarding the interface name and the comment that it might get confused with `org.eclipse.hono.client.adapter.command.CommandConsumerFactory`:
I thought about naming it `CommandRouterCommandConsumerFactory` instead, but then the name for the implementation class would get even longer still. 
As the Command Router module isn't added as dependency in any other module, the confusion could only arise in the Command Router module itself. But I don't think it's much of an issue there.

Reply:
ok